siriuspy/siriuspy/bsmp/entities.py

```diff
@@ -1,34 +1,49 @@
 """BSMP entities."""
 import struct as _struct
+import typing
 
 import numpy as _np
 
+from .types import BSMPType
+
 
 class Entity:
     """BSMP entity."""
 
-    def _conv_value(self, fmt, value):
+    def _conv_value(self, fmt: str, value) -> typing.List[str]:
         if fmt == '<c':
             return value
         else:
             return [chr(c) for c in _struct.pack(fmt, value)]
 
-    def _check_type(self, var_type, value):
+    def _check_type(self, var_type: BSMPType, value: typing.Union[str, int, float]):
         if not var_type.check(value):
             raise TypeError("{}, {}".format(var_type.type, value))
 
-    def _calc_types_size(self, var_types):
+    def _calc_types_size(
+        self,
+        var_types: typing.Tuple[BSMPType],
+    ) -> int:
         size = 0
         try:
             for var_type in var_types:
                 size += var_type.size
         except TypeError:
-            return var_types.size
+            if isinstance(var_types, BSMPType):
```

Comment:
@xresende  here I am including another check, so we don't assume the user is passing the correct type 
```python
BSMPType iterable | BSMPType
```